[browser] rename C symbols

pavelsavara opened the pull request:
There are C symbols that are used by PAL or ICall and are implemented in browser specific C or JavaScript. 
Many of them would be re-used by coreCLR and are also already used by NAOT_LLVM.

```c
// System
SystemJS_ConsoleClear
SystemJS_GetCurrentProcessId
SystemJS_GetLocaleInfo
SystemJS_RandomBytes
SystemJS_ScheduleBackgroundJob
SystemJS_ScheduleTimer
SystemJS_WarnAboutBlockingWait

// ST interop
SystemJSInterop_AssemblyGetEntryPoint
SystemJSInterop_BindAssemblyExports
SystemJSInterop_BindJSImportST
SystemJSInterop_CancelPromise
SystemJSInterop_UnregisterGCRoot
SystemJSInterop_GetAssemblyExport
SystemJSInterop_InvokeJSFunction
SystemJSInterop_InvokeJSImportST
SystemJSInterop_RegisterGCRoot
SystemJSInterop_ReleaseCSOwnedObject
SystemJSInterop_ResolveOrRejectPromise

// MT interop
SystemJSInterop_CancelPromisePost
SystemJSInterop_InstallWebWorkerInterop
SystemJSInterop_InstallWebWorkerInterop
SystemJSInterop_InvokeJSFunctionSend
SystemJSInterop_InvokeJSImportAsyncPost
SystemJSInterop_InvokeJSImportSync
SystemJSInterop_InvokeJSImportSyncSend
SystemJSInterop_ResolveOrRejectPromisePost
SystemJSInterop_ReleaseCSOwnedObjectPost
SystemJSInterop_UninstallWebWorkerInterop
```

pavelsavara committed:
rename C symbols

pavelsavara committed:
more

pavelsavara commented:
> I don't see much value in `dotnet_` prefix, as all of it is internal (IIUC)

In case there are 3rd party libraries linked into the same .wasm file, we don't want to clash.

As I'm working on the host, the idea of `BrowserNative_RandomBytes` and `src\native\libs\Browser.Native` makes more and more sense.

SingleAccretion commented:
I think functions like `mono_wasm_browser_entropy` should be prefixed by the name of their respective native shim (like `SystemNative_GetBrowserEntropy`, there are also some globalization-related ones). Actually, I would like to see these small helpers implemented as Emscripten JS libraries / EM_JS for linking and layering reasons, but that's a separate conversation.

For the JS interop ones, I don't have a strong opinion (they could be prefixed with `SystemJSInterop` I guess...).

pavelsavara committed:
Merge branch 'main' into mono_wasm_prefix

pavelsavara committed:
update to follow existing style if native exports

pavelsavara commented:
Changed to `SystemJS_*` and `SystemJSInterop_*`

jkotas commented on src/mono/browser/runtime/exports-binding.ts:
```diff
@@ -44,15 +44,15 @@ export const mono_wasm_threads_imports = !WasmEnableThreads ? [] : [
-    mono_wasm_invoke_jsimport_MT,
-    mono_wasm_warn_about_blocking_wait,
+    SystemJSInterop_InstallWebWorkerInteropJs,
+    SystemJSInterop_UninstallWebWorkerInterop,
```
Should both Install and Uninstall have the same suffix? (InteropJs vs. Interop)

If you keep `Js` in the name, it may want to be capitalized as `InteropJS` to match `SystemJS`.

pavelsavara replied:
- `SystemJSInterop_InstallWebWorkerInterop` is C which wraps `SystemJSInterop_InstallWebWorkerInteropJs` 
- `SystemJSInterop_UninstallWebWorkerInterop` is mapped from JS directly to ICall. In a way, `SystemJSInterop_InstallWebWorkerInteropJs` is implementation detail.

pavelsavara replied:
capitalized JS suffix

pavelsavara committed:
feedback: capitalize JS suffix

pavelsavara committed:
- revert mono_wasm_schedule_timer_tick

- fix SystemJSInterop_CancelPromiseImpl
- JS -> Impl suffix
- SystemJSInterop_UnregisterGCRoot
- fix rollup.config

pavelsavara committed:
revert

radekdoulik commented on src/native/minipal/random.c:
```diff
@@ -78,11 +78,11 @@ int32_t minipal_get_cryptographically_secure_random_bytes(uint8_t* buffer, int32
 
 #ifdef __EMSCRIPTEN__
-    extern int32_t mono_wasm_browser_entropy(uint8_t* buffer, int32_t bufferLength);
+    extern int32_t SystemJS_RandomBytes(uint8_t* buffer, int32_t bufferLength);
```
nm, I missed it above

radekdoulik reviewed (approved):
LGTM

maraf reviewed (commented):
Looks good to me in general. One question related to the naming:

The C# part is S.R.InteropServices.JavaScript, shouldn't the unmanaged prefix be SystemInteropJS?

pavelsavara committed:
Merge branch 'main' into mono_wasm_prefix

pavelsavara committed:
feedback SystemJSInterop_ -> SystemInteropJS_
